collector_test.go

```diff
@@ -6,9 +6,90 @@ import (
 
 	"github.com/soniah/gosnmp"
 
+	"github.com/prometheus/client_model/go"
 	"github.com/prometheus/snmp_exporter/config"
 )
 
+func TestPduToSample(t *testing.T) {
+	cases := []struct {
+		pdu            *gosnmp.SnmpPDU
+		indexOids      []int
+		metric         *config.Metric
+		oidToPdu       map[string]gosnmp.SnmpPDU
+		expectedMetric string
+	}{
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.1.1.1.1",
+				Type:  0x02,
+				Value: 2,
+			},
+			indexOids: []int{},
+			metric: &config.Metric{
+				Name: "test_metric",
+				Oid:  "1.1.1.1.1",
+				Type: "counter",
+				Help: "Help string",
+			},
+			oidToPdu:       make(map[string]gosnmp.SnmpPDU),
+			expectedMetric: "counter:<value:2 > ",
+		},
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.1.1.1.1",
+				Type:  0x02,
+				Value: 2,
+			},
+			indexOids: []int{},
+			metric: &config.Metric{
+				Name: "test_metric",
+				Oid:  "1.1.1.1.1",
+				Type: "gauge",
+				Help: "Help string",
+			},
+			oidToPdu:       make(map[string]gosnmp.SnmpPDU),
+			expectedMetric: "gauge:<value:2 > ",
+		},
+		{
+			pdu: &gosnmp.SnmpPDU{
+				Name:  "1.1.1.1.1",
+				Type:  0x02,
+				Value: -2,
+			},
+			indexOids: []int{},
+			metric: &config.Metric{
+				Name: "test_metric",
+				Oid:  "1.1.1.1.1",
+				Help: "Help string",
+			},
+			oidToPdu:       make(map[string]gosnmp.SnmpPDU),
+			expectedMetric: "label:<name:\"test_metric\" value:\"-2\" > gauge:<value:1 > ",
+		},
+	}
+
+	for _, c := range cases {
+		m := pduToSample(c.indexOids, c.pdu, c.metric, c.oidToPdu)
+		metric := &io_prometheus_client.Metric{}
+		err := m.Write(metric)
+		if err != nil {
+			t.Fatalf("Error writing metric: %v", err)
+		}
+		if metric.String() != c.expectedMetric {
+			t.Fatalf("Unexpected metric: got %v, want %v", metric.String(), c.expectedMetric)
+		}
+	}
+}
+
+func TestGetPduValue(t *testing.T) {
+	pdu := &gosnmp.SnmpPDU{
+		Value: -2,
```

Comment:
Signed integers are supported by SNMP, it's the Counter64 which are the uint64s that we need to be careful with.